### ISSUE TITLE
Ensure the new DatePicker can be used in WTFDateField etc.

### ISSUE THREAD

**Issue by ThiefMaster** [indico/indico#6662]
I started [this branch](https://github.com/ThiefMaster/indico/tree/wtf-date-picker) where I adapted the WTFDateField component to work with the new date picker, and also added some test code so the form on `/admin/announcement` has two linked date fields for easier testing purposes.

However, right now there are various styling issues, which I think are related to the general CSS from Indico which if course also affects the date picker...

- The field has a max width like all inputs, but the trigger button is outside. Setting `ind-date-picker { width: fit-content; }` kind of fixes it (maybe there's a better fix) but the trigger button is still misaligned. ![image](https://github.com/user-attachments/assets/f24bfea2-5c28-40eb-9fb5-e0fe544ef4af)
- The calendar popup is unnecessarily big, because field styling leaks into it: 
![image](https://github.com/user-attachments/assets/4cddb163-1a57-4f74-b255-f8a06f9c3502)
- When the field is configured to be clearable, that button is misaligned. Likely an easy fix, but I wonder if it wouldn't be nice to have in the `DatePicker` itself instead of having to do it from outside (I think it might make sense to allow easy clearing whenever the field is not required): 
![image](https://github.com/user-attachments/assets/8bf3f0fc-49e9-4a0e-8c22-fa48f636b81a)

**Comment by foxbunny** [comment]
I don't have those fields in thee announcements. Is that some plugin?

![image](https://github.com/user-attachments/assets/b1c19269-2482-45e9-bf47-23d5cc8448ff)

**Comment by ThiefMaster** [comment]
check thr branch i linked in the issue, i just added them there for testing purposes since it's a convenient place to test the fields (no extra clicks to open dialogs etc)

**Comment by ThiefMaster** [comment]
I also just rebased that one to your latest `dual-calendar` branch

**Comment by ThiefMaster** [comment]
@foxbunny it might be a good opportunity to revive this one :)

I rebased [my branch](https://github.com/ThiefMaster/indico/tree/wtf-date-picker) to the latest `dual-calendar` once again (since that one's probably going to be merged soon anyway).

**Comment by foxbunny** [comment]
After two days of battling with this, here's the findings thus far.

I'm 90% there, but the date picker control is found in a nasty cross-fire between SUI and the `_form.scss` design system. For some reason, the `_inputs.scss` (used by `_form.scss`) references SUI's `.ui` class, which is giving me a major headache.

I think we have 2 realistic options here:

1. Cleanly separate the old (?) inputs/forms partials from SUI so they don't reference each other in any shape or form, and then revisit this, and (optionally) stop loading the partials where SUI is used.

2. Forcing values with `!important`. Generally not a big fan of adding debt to brand new code, but since it's already done multiple times, I'm not categorically opposed.

What do you think?

**Comment by ThiefMaster** [comment]
I think (2) is the only viable option... :/

**Comment by foxbunny** [comment]
Here's the patch I came up with. For some reason, your fork doesn't appear in the list of PR targets, so I'm giving you a link to the commit instead.

https://github.com/foxbunny/indico/commit/5987f6da1df2956fe5a3ccebdda4ae0dd46f27a3

**Comment by ThiefMaster** [comment]
It looks like it breaks the one in room booking :/

![Image](https://github.com/user-attachments/assets/808a4f58-0a93-4a67-93a5-13c5dc17d754)

**Comment by ThiefMaster** [comment]
But the WTForms version looks fine now :)

**Comment by ThiefMaster** [comment]
PS: I updated my branch to use more recent dates in the two test fields so allow for some actual testing

**Comment by ThiefMaster** [comment]
All the wtforms versions should be contained inside a `form.i-form`. So you might be able to use that class to apply the different styling.

**Comment by foxbunny** [comment]
> It looks like it breaks the one in room booking :/

Ugh. So we have at least three different contexts for our widgets... How the... ok I won't ask. 🙈

FYI: from my experience thus far, liberal use of nesting is one of the biggest issues. Even the more complex selectors I use 'lose' to the existing deeply nested ones.

I think we might need to reboot the whole thing and go with shadow DOM after all. Usually it's not needed, but this is becoming a bit too flaky, and some style encapsulation would help us a great deal. Otherwise we'd be putting out fires like this every time we have a new widget. I'll try a few things tomorrow and we'll see how it goes.

**Comment by foxbunny** [comment]
Actually we didn't have 3 different contexts, just two: the range picker was consistently broken in both reg form and RB. That's good. At least we can barely get away with current approach, supposing that WTF is going to be phased out at some point. 🤞

Here's the fixed version:

https://github.com/foxbunny/indico/commit/91be78461371e1af1ebea8d98e3c7e94100634c5

**Comment by ThiefMaster** [comment]
thanks!

just some small styling weirdness (gray background of the month select dropdown)

![Image](https://github.com/user-attachments/assets/4f6dc053-ffd4-42aa-b547-1d287c029199)

PS: I rebased my wtf-date-picker branch to the latest master again

**Comment by foxbunny** [comment]
Here's the updated version:

https://github.com/foxbunny/indico/commit/bd69e42966b8c772a5756041c47e3ea49252a9f1
